src/client_side_request.cc

```diff
@@ -2057,8 +2057,13 @@ ClientHttpRequest::noteMoreBodyDataAvailable(BodyPipe::Pointer)
         bpc.checkIn();
     }
 
-    if (adaptedBodySource->exhausted())
+    if (adaptedBodySource->exhausted()) {
+        // XXX: Setting receivedWholeAdaptedReply here is a workaround for a
+        // regression, as described in https://bugs.squid-cache.org/show_bug.cgi?id=5187#c6
+        receivedWholeAdaptedReply = true;
+        debugs(85, DBG_IMPORTANT, "WARNING: Squid bug 5187 workaround triggered");
         endRequestSatisfaction();
```

Comment:
Since we do not have a proper fix, and others report success with using this small workaround, I am OK with this workaround going in, even though it may, in some cases, resurrect problems fixed in commit ba3fe8d.

I fixed PR description and am approving this PR, but I certainly do not insist that merging this workaround is the right path forward. @yadij, @kinkie, please block or remove merge clearance if you think this should not go in.